tools/selinux/icinga2.te

```diff
@@ -192,6 +193,9 @@ allow icinga2_t icinga2_port_t:tcp_socket name_bind;
 allow icinga2_t self:tcp_socket create_stream_socket_perms;
 corenet_tcp_connect_icinga2_port(icinga2_t)
 
+allow icinga2_t redis_port_t:tcp_socket name_connect;
+allow icinga2_t mysqld_db_t:sock_file write;
+
 mysql_stream_connect(icinga2_t)
```

Comment:
`mysql_stream_connect` sounds like this should already cover Unix domain sockets ([see also](http://oss.tresys.com/docs/refpolicy/api/contrib_mysql.html#link_mysql_stream_connect), not sure if this is a good reference though)?

Reply:
Well, that's what I was expecting, but it doesn't seem to work the way it should. I always just get the `(13 Permission denied)` error message.

```bash
[root@yh-centos mysql]# ls -alZ
...
srwxrwxrwx. mysql mysql system_u:object_r:mysqld_db_t:s0 mysql.sock
...

```

```bash
[2022-07-18 08:50:43 +0000] critical/IdoMysqlConnection: Connection to database 'icinga2' with user 'icinga2' on 'localhost:3306' failed: "Can't connect to local MySQL server through socket '/var/lib/mysql/mysql.sock' (13 "Permission denied")"
```

Reply:
https://dev.mysql.com/doc/refman/8.0/en/problems-with-mysql-sock.html answers part of my question (but why?):

> The default location for the Unix socket file that the server uses for communication with local clients is `/tmp/mysql.sock`. (For some distribution formats, the directory might be different, such as `/var/lib/mysql` for RPMs.)

Reply:
> But with your change, it works?

Yes.

> What distribution are you testing on? Is that the default path there?

I'm using centos 7 and yes, it is the default path.

Reply:
> https://dev.mysql.com/doc/refman/8.0/en/problems-with-mysql-sock.html answers part of my question (but why?):

Changing the file context to `mysqld_var_run_t` will also help.

```bash
semanage fcontext -a -t mysqld_var_run_t "/var/lib/mysql/mysql.sock"
restorecon -Rv /var/lib/mysql/mysql.sock
```

```
[root@yh-centos icinga2-selinux-2.13.4]# ls -alZ /var/lib/mysql/
srwxrwxrwx. mysql mysql system_u:object_r:mysqld_var_run_t:s0 mysql.sock
```